app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,20 @@
+class CommentsController < ApplicationController
+  def create
+    product = Product.find_by(id: params[:comment][:product_id])
+    if !logged_in?
+      flash[:danger] = t "base.request_login"
```

Comment:
Cái này cần ko chú,  nên xử lý chưa login hoàn toàn ở method check login đi, ở đây chỉ cần dùng action filter để gọi thôi